tenable/io/v3/vm/agents/api.py

```diff
@@ -0,0 +1,124 @@
+'''
+Agents
+======
+
+The following methods allow for interaction into the Tenable.io
+:devportal:`agents <agents>` API endpoints.
+
+Methods available on ``tio.v3.vm.agents``:
+
+.. rst-class:: hide-signature
+.. autoclass:: AgentsAPI
+    :members:
+'''
+from typing import Dict, Union
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.vm.agents.schema import AgentSchema
+
+
+class AgentsAPI(ExploreBaseEndpoint):
+    '''
+    This class contains methods related to Agents API
+    '''
+    _path: str = 'api/v3/agents'
+    _conv_json: bool = True
+    _schema = AgentSchema()
+
+    def details(self, agent_id: UUID) -> Dict:
+        '''
+        Retrieves the details of an agent.
+
+        :devportal:`agents: get <agents-get>`
+
+        Args:
+            agent_id (UUID): The unique identifier of the agent.
+
+        Returns:
+            :obj:`dict`:
+                The agent dictionary record.
+
+        Examples:
+            >>> agent = tio.v3.vm.agents.details(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+            >>> pprint(agent)
+        '''
+        return self._get(f'{agent_id}')
+
+    def list_agents_from_group(self):
+        raise NotImplementedError(
+            "This method will be implemented later."
+        )
+
+    def task_status(self, task_uuid: UUID) -> Dict:
```

Comment:
This is not a publicly documented API endpoint, so we can remove it here

Reply:
That's correct. It is not publicly available under the agent API. Though, it is available under the agent bulk operation API documentation. Moreover, this function was also present in the current SDK, so we have included it in v3 as well.
Ref: https://developer.tenable.com/reference#bulk-task-agent-status

Please let us know if you still want us to remove this. In this case, should we include it in the Agent Bulk Operations module of patentable that we add in the future?

Reply:
Ok, we can keep it if it is documented. I must have missed that.